packages/Popover/src/Popover.js

```diff
@@ -305,29 +302,25 @@ class Popover extends React.Component {
 
   handleKeyDown = event => {
     if (event.key === "Tab" && this.isOpen() && this.$trigger) {
-      if (event.shiftKey && this.focusIsOnFirstFocusableElementInPopover()) {
+      const isFocusOnFirst = this.focusIsOnCertainElementInPopover("first") || document.activeElement === this.$content;
+      const isFocusOnLast = this.focusIsOnCertainElementInPopover("last");
+
+      if (event.shiftKey && isFocusOnFirst) {
         event.preventDefault();
-        focusableElements[triggerFocusIndex].focus(); // this closes the popover, which restores tabIndexes
-      } else if (!event.shiftKey && this.focusIsOnLastFocusableElementInPopover()) {
+        this.restoreTabIndexes();
+        this.focusableElements[this.triggerFocusIndex].focus(); // this closes the popover, which restores tabIndexes
+      } else if (!event.shiftKey && isFocusOnLast) {
         event.preventDefault();
-        focusableElements[triggerFocusIndex + 1].focus(); // this closes the popover, which restores tabIndexes
+        this.restoreTabIndexes();
+        this.focusableElements[this.triggerFocusIndex + 1].focus(); // this closes the popover, which restores tabIndexes
```

Comment:
Oops, forgot to remove the comment... but in the case with controlled Popover with no `<Popover.Trigger>`, it wasn't being restored, but it works when it's done explicitly above.  Not sure why.